WordPress/WordPressStatsWidgets/Helpers/WidgetDataReader.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+
+final class WidgetDataReader<T: HomeWidgetData> {
```

Comment:
I appreciate what is done with this reader, reusing common functionality for the providers. Do you see any more opportunities for such reuse? Would they make sense in your mind?

Reply:
Yes, theoretically all duplicated code could be extracted for reuse using a similar approach. I was thinking about what the reasonable responsibility of the timeline provider is.

As a final decision, I decided to leave the refresh time and legacy data check to fetch the API in the timeline provider, focusing on the refresh frequency and the API data source, does it make sense to you?

Reply:
> As a final decision, I decided to leave the refresh time and legacy data check to fetch the API in the timeline provider, focusing on the refresh frequency and the API data source, does it make sense to you?

I looked through the code, I think it's a reasonable balance 👍